website/util/__init__.py

```diff
@@ -28,6 +29,30 @@
     'create_folder': 'file',
 }
 
+# Function courtesty of @brianjgeiger, moved from API utils
+def rapply(collection, func, *args, **kwargs):
+    """Recursively apply a function to all values in a dict or list
+
+    :param dict | list collection:
+    :param function func:
+    """
+    if isinstance(collection, collections.Mapping):
+        return {
+            key: rapply(value, func, *args, **kwargs)
+            for key, value in collection.iteritems()
+        }
+    elif isinstance(collection, list):
```

Comment:
Not sure if this is a *good* idea, but the below would consolidate things, and appears to pass all tests for lists, tuples, sets, dicts, and strings.. also, the docstring may wish to note that this can operate on strings. (by design)

```
def rapply(collection, func, *args, **kwargs):
    """Recursively apply a function to all values in a dict or list
    :param dict | list collection:
    :param function func:
   """
    if isinstance(data, collections.Mapping):
        return {
            key: rapply(value, func, *args, **kwargs)
            for key, value in data.iteritems()
        }
    elif isinstance(data, collections.Iterable) and not isinstance(data, basestring):
        desired_type = type(data)
        return desired_type(
            rapply(item, func, *args, **kwargs) for item in data
        )
    else:
        return func(data, *args, **kwargs)
```

And tests
```

    def test_rapply_on_set(self):
        inputs = set(i for i in range(5))
        add_one = lambda n: n + 1
        outputs = util.rapply(inputs, add_one)
        for i in inputs:
            assert_in(i + 1, outputs)
        assert_true(isinstance(outputs, set))

    def test_rapply_on_str(self):
        input = "bob"
        convert = lambda s: s.upper()
        outputs = util.rapply(input, convert)

        assert_equal("BOB", outputs)
        assert_true(isinstance(outputs, basestring))
```